### PR TITLE
Operators for double int and string propeties

### PR DESCRIPTION
**Pull request by jensschmeink**
Basic implementation of some operators for +=, -=, ... 
These are now implemented for double, int and string property

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

### CODE REVIEW

**Comment by jensschmeink** [comment]
Basic unittests are needed but currently missing

File: src/dataprocessor/property/gt_doubleproperty.cpp [hunk 1]
```diff
@@ -137,6 +137,35 @@ GtDoubleProperty::GtDoubleProperty(const QString& ident,
...
+GtDoubleProperty::operator*=(const double& b)
+{
+    double newVal = getVal() * b;
+    setVal(newVal);
+}
+
+void
+GtDoubleProperty::operator/=(const double& b)
+{
+    assert(a != 0.0);
```
Comment (jensschmeink) [inline src/dataprocessor/property/gt_doubleproperty.cpp:164]:
```suggestion
    assert(b != 0.0);
```

File: src/dataprocessor/property/gt_doubleproperty.h [hunk 2]
```diff
@@ -99,6 +99,13 @@ class GT_DATAMODEL_EXPORT GtDoubleProperty : public GtProperty<double>
                      const double boundary,
                      const double& value = 0.0);
 
+    // operator overloads
+    inline void operator+=(double const& b);
+    inline void operator-=(double const& b);
+    inline void operator*=(double const& b);
+    inline void operator/=(double const& b);
```
Comment (jensschmeink) [inline src/dataprocessor/property/gt_doubleproperty.h:106]:
```suggestion
    void operator+=(double const& b);
    void operator-=(double const& b);
    void operator*=(double const& b);
    void operator/=(double const& b);
```

File: src/dataprocessor/property/gt_intproperty.h [hunk 3]
```diff
@@ -124,6 +124,12 @@ class GT_DATAMODEL_EXPORT GtIntProperty : public GtProperty<int>
                   const int boundary,
                   const int& value = 0.0);
 
+    // operator overloads
+    inline void operator+=(int b);
+    inline void operator-=(int b);
+    inline void operator*=(int b);
+    inline void operator/=(int b);
```
Comment (jensschmeink) [inline src/dataprocessor/property/gt_intproperty.h:131]:
```suggestion
    void operator+=(int b);
    void operator-=(int b);
    void operator*=(int b);
    void operator/=(int b);
```

File: src/dataprocessor/property/gt_stringproperty.h [hunk 4]
```diff
@@ -50,6 +50,9 @@ class GT_DATAMODEL_EXPORT GtStringProperty : public GtProperty<QString>
                      const QString& value = QString(),
                      QValidator* validator = nullptr);
 
+    // operator overloads
+    inline void operator+=(const QString& b);
```
Comment (jensschmeink) [inline src/dataprocessor/property/gt_stringproperty.h:54]:
```suggestion
    void operator+=(const QString& b);
```

**Review by rainman110 — CHANGES_REQUESTED** [review]
Looks good, but could you fixup the indentation error in the tests?

File: tests/unittests/datamodel/test_gt_doubleproperty.cpp [hunk 5]
```diff
@@ -385,3 +385,36 @@ TEST_F(TestGtDoubleProperty, optional)
...
+	
+	prop *= factor;
+    ASSERT_DOUBLE_EQ(prop.get(), 3.3 * factor);
+	
+	// operator/=
+	double div = 4.4;
+    prop.setVal(3.3);
+	
+	prop /= div;
+    ASSERT_DOUBLE_EQ(prop.get(), 3.3 / div);
```
Comment (jensschmeink) [inline tests/unittests/datamodel/test_gt_doubleproperty.cpp:419]:
```suggestion
    GtDoubleProperty prop("prop", "test", "test", GtUnit::Category::None, 4.0);
	
    // operator+=
    double sum = 4.4;
    prop.setVal(3.3);
	
    prop += sum;
    ASSERT_DOUBLE_EQ(prop.get(), 3.3 + sum);
	
    // operator-=
    double diff = 4.4;
    prop.setVal(3.3);
	
    prop -= diff;
    ASSERT_DOUBLE_EQ(prop.get(), 3.3 - diff);
	
    // operator*=
    double factor = 4.4;
    prop.setVal(3.3);
	
     prop *= factor;
    ASSERT_DOUBLE_EQ(prop.get(), 3.3 * factor);
	
    // operator/=
    double div = 4.4;
    prop.setVal(3.3);
	
    prop /= div;
    ASSERT_DOUBLE_EQ(prop.get(), 3.3 / div);
```

File: tests/unittests/datamodel/test_gt_intproperty.cpp [hunk 6]
```diff
@@ -307,3 +307,35 @@ TEST_F(TestGtIntProperty, optional)
...
+	
+	prop *= factor;
+    EXPECT_EQ(prop.get(), 13 * factor);
+	
+	// operator/=
+	int div = 4;
+    prop.setVal(13);
+	
+	prop /= div;
+    EXPECT_EQ(prop.get(), 3);
```
Comment (jensschmeink) [inline tests/unittests/datamodel/test_gt_intproperty.cpp:340]:
```suggestion
    GtIntProperty prop("prop", "test", "test", 4);
    // operator+=
    int sum = 4;
    prop.setVal(13);
	
    prop += sum;
    EXPECT_EQ(prop.get(), 13 + sum);

    // operator-=
    int diff = 4;
    prop.setVal(13);
	
    prop -= diff;
    EXPECT_EQ(prop.get(), 13 - diff);
	
    // operator*=
    int factor = 4;
    prop.setVal(13);
	
    prop *= factor;
    EXPECT_EQ(prop.get(), 13 * factor);
	
    // operator/=
    int div = 4;
    prop.setVal(13);
	
    prop /= div;
    EXPECT_EQ(prop.get(), 3);
```

File: tests/unittests/datamodel/test_gt_stringproperty.cpp [hunk 7]
```diff
@@ -135,3 +135,15 @@ TEST_F(TestGtStringProperty, optional)
...
+TEST_F(TestGtStringProperty, extendenOperator)
+{
+	GtStringProperty prop("prop", "test", "test", "hello");
+	
+	// operator+=
+	QString added = " is nice!";
+    prop.setVal("GTlab");
+	
+	prop += added;
+    ASSERT_STREQ(prop.get().toStdString().c_str(), "GTlab is nice!");
```
Comment (jensschmeink) [inline tests/unittests/datamodel/test_gt_stringproperty.cpp:148]:
```suggestion
    GtStringProperty prop("prop", "test", "test", "hello");
	
    // operator+=
    QString added = " is nice!";
    prop.setVal("GTlab");
	
    prop += added;
    ASSERT_STREQ(prop.get().toStdString().c_str(), "GTlab is nice!");
```

**Comment by jensschmeink** [comment]
> Looks good, but could you fixup the indentation error in the tests?

Done

File: tests/unittests/datamodel/test_gt_doubleproperty.cpp [hunk 8]
```diff
@@ -385,3 +385,36 @@ TEST_F(TestGtDoubleProperty, optional)
...
+    prop.setVal(3.3);
+	
+    prop -= diff;
+    ASSERT_DOUBLE_EQ(prop.get(), 3.3 - diff);
+	
+    // operator*=
+    double factor = 4.4;
+    prop.setVal(3.3);
+	
+     prop *= factor;
```
Comment (rainman110) [inline tests/unittests/datamodel/test_gt_doubleproperty.cpp:411]: I don't want to be picky, but there's still a indentation error ;)

Comment (jensschmeink) [inline tests/unittests/datamodel/test_gt_doubleproperty.cpp:411]:
```suggestion
    prop *= factor;
```

**Review by rainman110 — APPROVED** [review]

**Comment by rainman110** [comment]
@jensschmeink Please stash before merge

**Pull request merged**